File: server/DataService/src/main/java/ru/hse/online/storage/PathData.java
package ru.hse.online.storage;

import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Data;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.*;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
@Table("paths")
public class PathData {
    @NotNull
    @PrimaryKey
    private final PathKey key;

    @NotNull
    @Column("polyline")
    private final String polyline;

    @NotNull
    @Column("created")
    private final LocalDate created;

    @Column("name")
    private final String name;

    @Column("distance")
    private final Double distance;

    @Column("duration")
    private final Double duration;

    @Data
    @Builder
    @PrimaryKeyClass
    public static class PathKey implements Serializable {
        @PrimaryKeyColumn(name = "user_id", type = PrimaryKeyType.PARTITIONED)
        private final UUID userId;

        @PrimaryKeyColumn(name = "path_id", type = PrimaryKeyType.CLUSTERED)
        private final UUID pathId;
    }
}

File: client/app/src/main/java/ru/hse/online/client/repository/networking/api_data/PathData.kt
package ru.hse.online.client.repository.networking.api_data

import com.google.gson.annotations.JsonAdapter
import ru.hse.online.client.repository.networking.adapter.LocalDateAdapter
import java.time.LocalDate
import java.util.*

data class PathRequest(
    val userId: UUID,
    val polyline: String,
    @JsonAdapter(LocalDateAdapter::class)
    val created: LocalDate,
    val name: String,
    val distance: Double,
    val duration: Double
)

data class PathResponse(
    val userId: UUID,
    val pathId: UUID,
    val polyline: String,
    @JsonAdapter(LocalDateAdapter::class)
    val created: LocalDate,
    val name: String,
    val distance: Double,
    val duration: Double
)

sealed class PathResult {
    data class Success(
        val paths: List<PathResponse>? = null,
        val code: Int
    ) : PathResult()

    data class Failure(
        val code: Int? = null,
        val message: String?
    ) : PathResult()
}